File: client/src/app/modules/miscellaneous/services/language.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Language } from '../../miscellaneous/model/Language';

@Injectable({
  providedIn: 'root'
})
export class LanguageService extends AppRestService<Language>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getLanguages() {
    return this.getListCached(new HttpParams(), "languages");
  }

  addOrUpdateLanguage(language: Language) {
    this.clearListCache(new HttpParams(), "languages");
    return this.addOrUpdate(new HttpParams(), "language", language, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/miscellaneous/service/LanguageService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.Language;

public interface LanguageService {
    public List<Language> getLanguages();

    public Language getLanguage(Integer id);
	
	 public Language addOrUpdateLanguage(Language language);
}
